### PR TITLE
Commit of full solution with respective tests

### PR DESCRIPTION
**Pull request by Mattaus-Isenberg**

### CODE REVIEW

**Comment by Mattaus-Isenberg** [comment]
- [x] Create Node Class
- [x] Create Stack Class
- [x] Create Queue Class
- [x] Created fully functional push(), pop(), peek(), isEmpty(), enqueue(), dequeue() methods.
- [x] Created all required tests for the classes and methods implemented

File: code401challenges/src/test/java/stacksandqueues/QueueTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+public class QueueTest
+{
+
+    private Object NullPointerException;
+
+    @Test
+    public void enqueueTest()
+    {
+        Queue expected = new Queue();
```
Comment (jeffborda) [inline code401challenges/src/test/java/stacksandqueues/QueueTest.java]: Remember when you make a generic data structure, declare it with the type in the diamond, like: `Queue<Integer> expected = new Queue<>();`

File: code401challenges/src/test/java/stacksandqueues/StackTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import static org.junit.Assert.*;
+
+public class StackTest {
+
+    private Object NullPointerException;
+
+    @Test
+    public void pushTest()
+    {
+        Stack expected = new Stack();
```
Comment (jeffborda) [inline code401challenges/src/test/java/stacksandqueues/StackTest.java]: Same with the Stack here, should be: `Stack<Integer> expected = new Stack<>();`

File: code401challenges/src/test/java/stacksandqueues/StackTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        expected.push(4);
+        expected.push(5);
+        expected.pop();
+        expected.pop();
+        expected.pop();
+        expected.pop();
+        expected.pop();
+        System.out.println(expected.top.node_Data);
+        assertEquals(NullPointerException, expected.top.node_Data);
+    }
```
Comment (jeffborda) [inline code401challenges/src/test/java/stacksandqueues/StackTest.java]: This test is failing. When you expect an exception, you don't need an assert statement, just put this next to the Test annotation:
@Test(expected = NullPointerException.class)

Then inside the test, just do whatever is going to cause the error.

**Pull request merged**